### PR TITLE
TYP: Annotations in pandas/core/{accessor, sorting}.py

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/accessor.py [hunk 1]
```diff
@@ -107,7 +109,9 @@ def f(self, *args, **kwargs):
                 setattr(cls, name, f)
 
 
-def delegate_names(delegate, accessors, typ, overwrite=False):
+def delegate_names(
+    delegate, accessors: Union[List[str], Set[str]], typ: str, overwrite: bool = False
```
Comment (jbrockmendel) [inline pandas/core/accessor.py]: @simonjayhawkins is there some container-ish thing we would use for `Union[List[foo], Set[foo]]`?  Also do we have a convention about annotating return from `__init__`?

File: pandas/core/accessor.py [hunk 2]
```diff
@@ -162,7 +166,7 @@ class CachedAccessor:
         the single argument ``data``.
     """
 
-    def __init__(self, name, accessor):
+    def __init__(self, name, accessor) -> None:
```
Comment (jbrockmendel) [inline pandas/core/accessor.py]: can we say anything about name?  str?  Optional[Hashable]?

File: pandas/core/sorting.py [hunk 3]
```diff
@@ -273,7 +275,7 @@ class _KeyMapper:
     Ease my suffering. Map compressed group id -> key tuple
     """
 
-    def __init__(self, comp_ids, ngroups, levels, labels):
+    def __init__(self, comp_ids, ngroups, levels, labels) -> None:
```
Comment (jbrockmendel) [inline pandas/core/sorting.py]: ngroups seems likely to be int, the others could probably be figured out without too much effort

File: pandas/core/accessor.py (hunk 2 above)
Comment (MomIsBestFriend) [reply pandas/core/accessor.py]: ```str``` worked

File: pandas/core/sorting.py [hunk 4]
```diff
@@ -273,7 +275,7 @@ class _KeyMapper:
     Ease my suffering. Map compressed group id -> key tuple
     """
 
-    def __init__(self, comp_ids, ngroups, levels, labels):
+    def __init__(self, comp_ids, ngroups: int, levels, labels) -> None:
```
Comment (simonjayhawkins) [inline pandas/core/sorting.py]:
```suggestion
    def __init__(self, comp_ids, ngroups: int, levels, labels):
```

Comment (MomIsBestFriend) [reply pandas/core/sorting.py]: @simonjayhawkins Do you want to unannotate every ```__init__```?

Like removing the ```-> None``` from every` `` _init__```?

File: pandas/core/accessor.py [hunk 5]
```diff
@@ -58,7 +58,9 @@ def _delegate_method(self, name, *args, **kwargs):
         raise TypeError("You cannot call method {name}".format(name=name))
 
     @classmethod
-    def _add_delegate_accessors(cls, delegate, accessors, typ, overwrite=False):
+    def _add_delegate_accessors(
+        cls, delegate, accessors: List[str], typ: str, overwrite: bool = False
```
Comment (simonjayhawkins) [inline pandas/core/accessor.py]: the type annotation of `accessors` here needs to be compatible with `delegate_names` below.

`_add_delegate_accessors` is used in the inner function of `delegate_names`. This inner function `add_delegate_accessors` does not have type annotations and is therefore not checked by mypy. Annotating the inner functions gives `pandas\core\accessor.py:143: error: Argument 2 to "_add_delegate_accessors" of "PandasDelegate" has incompatible type "Union[List[str], Set[str]]"; expected "List[str]"`

Probably `Sequence[str]` is more appropriate, however beware a string is also a sequence of str.

```
from typing import Sequence


def foo(bar: Sequence[str]):
    pass


foo(3)
foo("hdfjjdhf")
foo([1, 2, 3])
foo(["bar", "baz"])
```
```
test.py:8: error: Argument 1 to "foo" has incompatible type "int"; expected "Sequence[str]"
test.py:10: error: List item 0 has incompatible type "int"; expected "str"
test.py:10: error: List item 1 has incompatible type "int"; expected "str"
test.py:10: error: List item 2 has incompatible type "int"; expected "str"
```

File: pandas/core/sorting.py (hunk 4 above)
Comment (simonjayhawkins) [reply pandas/core/sorting.py]: if you come across any cases of `__init__` returning `None`, you can remove the `None`.

File: pandas/core/accessor.py (hunk 5 above)
Comment (MomIsBestFriend) [reply pandas/core/accessor.py]: Every combination I tried I got an error. I think I will drop the annotations for ```accessors``` in this PR.

**Comment by jbrockmendel** [comment]
pending word from @simonjayhawkins on whetheer we bother annotating `-> None` for `__init__`, LGTM

**Comment by simonjayhawkins** [comment]
> pending word from @simonjayhawkins on whetheer we bother annotating `-> None` for `__init__`, LGTM

it's an undocumented convention that we're not annotating the return type of `__init__`

**Review by simonjayhawkins — APPROVED** [review]

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @MomIsBestFriend

**Pull request merged**